index.d.ts

```diff
@@ -0,0 +1,9 @@
+declare module 'ol/extent';
+declare module 'ol/geom/Geometry';
+declare module 'ol/style/Style' {
+    export = ol.style.Style;
+}
+declare module 'ol/geom/Polygon';
+declare module 'ol/geom/LineString';
+declare module 'ol/render/canvas/Immediate';
+declare module 'ol/transform';
```

Comment:
You don't need to declare these. Just ad `"declaration": true,` to the tsconfig.json and `"types": "dist/index.d.ts"` to the package.json. Compare https://github.com/terrestris/geostyler/commit/2516c4093c25b2def7bbaeeb6eda507a7e8e5dcb

Reply:
`"declaration": true` did not work for me unfortunately. I renamed the `index.d.ts` to a less confusing name.